### ISSUE TITLE
x86: fails to disassemble shrd or shld

### ISSUE THREAD

**Issue by LRGH** [bdcht/amoco#32]
``` python
>>> print(cpu_x86.disassemble('\x0f\xa5\xef'))
None
>>> print(cpu_x86.disassemble('\x0f\xac\xef\x01'))
None
>>> print(cpu_x86.disassemble('\x0f\xa4\x6d\xf8\x01'))
None
```

I would like to use something like

``` python
@ispec_ia32("*>[ {0f}{a4} /r ib(8) ]", mnemonic = "SHLD")
```

but it is not possible.
As mentioned in the documentation, the "variable length directive" should be "at the end of the FORMAT".
How are we supposed to parse these instructions without using ugly hacks?

**Comment by LRGH** [comment]
Proposed solution:

``` python
@ispec_ia32("*>[ {0f}{ad} /r ]", mnemonic = "SHRD")
@ispec_ia32("*>[ {0f}{a5} /r ]", mnemonic = "SHLD")
def ia32_rm32_op3cl(obj,Mod,RM,REG,data):
    op1,data = getModRM(obj,Mod,RM,data)
    op2 = env.getreg(REG,op1.size)
    obj.operands = [op1, op2, env.cl]
    obj.type = type_data_processing

@ispec_ia32("*>[ {0f}{ac} /r ]", mnemonic = "SHRD")
@ispec_ia32("*>[ {0f}{a4} /r ]", mnemonic = "SHLD")
def ia32_rm32_op3cst(obj,Mod,RM,REG,data):
    op1,data = getModRM(obj,Mod,RM,data)
    op2 = env.getreg(REG,op1.size)
    imm = data[0:8]
    obj.operands = [op1, op2, env.cst(imm.int(),8)]
    obj.bytes += pack(imm)
    obj.type = type_data_processing
```

I also need to modify `asm.py` by adding in `i_SHRD` and `i_SHLD`

``` python
  if not op3._is_cst:
    # op3 is 'cl'
    # cannot emulate
    # BUG
    return
```

**Comment by bdcht** [comment]
thanks, fixed by c84e20f
